plugins/org.jkiss.dbeaver.ui.dashboard/src/org/jkiss/dbeaver/ui/dashboard/registry/DashboardRegistry.java

```diff
@@ -162,21 +162,35 @@ public DashboardDescriptor getDashboard(String id) {
     }
 
     /**
-     * Find dashboard matchign source. Source can be {@link DBPDataSourceContainer}, {@link DBPDataSourceProviderDescriptor} or {@link DBPDriver}
+     * Find dashboard matching source.
+     * Source can be {@link DBPDataSourceContainer}, {@link DBPDataSourceProviderDescriptor} or {@link DBPDriver}
      */
     public List<DashboardDescriptor> getDashboards(DBPNamedObject source, boolean defaultOnly) {
+        if (source == null) {
+            return List.of();
+        }
         if (source instanceof DBPDataSourceContainer) {
             source = ((DBPDataSourceContainer) source).getDriver();
         }
-        String providerId, driverId, driverClass;
+        String providerId = null;
+        String driverId = null;
+        String driverClass = null;
         if (source instanceof DBPDataSourceProviderDescriptor) {
             providerId = ((DBPDataSourceProviderDescriptor) source).getId();
             driverId = null;
             driverClass = null;
-        } else {
-            providerId = ((DBPDriver)source).getProviderId();
-            driverId = ((DBPDriver)source).getId();
-            driverClass = ((DBPDriver)source).getDriverClassName();
+        } else if (source instanceof DBPDriver) {
+            DBPDriver dbpDriver = (DBPDriver) source;
+            DBPDataSourceProviderDescriptor parentProvider = dbpDriver.getProviderDescriptor().getParentProvider();
+            if (parentProvider != null) {
+                // For now all dashboards we have - not children providers.
+                // If dashboard code is in EE repository - then it will work only for EE, even if parent in CE.
```

Comment:
I don't get it. Could you elaborate? As I understand you claim that dashboards created in ee dbeaver can't be used if I run ce dbeaver later. Will such dashboards be like as an empty dashboard window or how does it look like?

Can't we make it work between editions with recent @ShadelessFox's fix about data source's ids?

Reply:
You can open a Dashboard for any connection, but it will be empty if plugin.xml has not dashboard extension point.